src/flask/blueprints.py

```diff
@@ -269,19 +269,25 @@ def register(self, app: "Flask", options: dict) -> None:
         :param first_registration: Whether this is the first time this
             blueprint has been registered on the application.
         """
-        first_registration = False
-
-        if self.name in app.blueprints:
-            assert app.blueprints[self.name] is self, (
-                "A name collision occurred between blueprints"
-                f" {self!r} and {app.blueprints[self.name]!r}."
-                f" Both share the same name {self.name!r}."
-                f" Blueprints that are created on the fly need unique"
-                f" names."
+        first_registration = True
+
+        for blueprint in app.blueprints.values():
+            if blueprint is self:
+                first_registration = False
+
+        name = (
+            f"{options.get('name_prefix', '')}.{options.get('name', self.name)}".lstrip(
+                "."
+            )
+        )
```

Comment:
This formatting is weird, extract the `get` calls to variables to keep it from wrapping.